bot/exts/evergreen/connect_four.py

```diff
@@ -0,0 +1,437 @@
+
+    def get_player(
+            self,
+            ctx: commands.Context,
+            announcement: discord.Message,
+            reaction: discord.Reaction,
+            user: discord.Member
+    ) -> bool:
+        """Predicate checking the criteria for the announcement message."""
+        if self.already_playing(ctx.author):  # If they've joined a game since requesting a player 2
+            return True  # Is dealt with later on
+
+        if (
+                user.id not in (ctx.me.id, ctx.author.id)
+                and str(reaction.emoji) == HAND_RAISED_EMOJI
+                and reaction.message.id == announcement.id
+        ):
+            if self.already_playing(user):
+                self.bot.loop.create_task(ctx.send(f"{user.mention} You're already playing a game!"))
+                self.bot.loop.create_task(announcement.remove_reaction(reaction, user))
+                return False
+
+            if user in self.waiting:
+                self.bot.loop.create_task(ctx.send(
+                    f"{user.mention} Please cancel your game first before joining another one."
+                ))
+                self.bot.loop.create_task(announcement.remove_reaction(reaction, user))
+                return False
+
+            return True
+
+        if (
+                user.id == ctx.author.id
+                and str(reaction.emoji) == CROSS_EMOJI
+                and reaction.message.id == announcement.id
+        ):
+            return True
+        return False
+
+    def already_playing(self, player: discord.Member) -> bool:
+        """Check if someone is already in a game."""
+        return any(player in (game.player1, game.player2) for game in self.games)
+
+    async def _play_game(
+            self,
+            ctx: commands.Context,
+            user: typing.Optional[discord.Member],
+            board_size: int,
+            emoji1: str,
+            emoji2: str
+    ) -> None:
+        """Helper for playing a game of connect four."""
+        global EMOJIS
+        EMOJIS = [":white_circle:", str(emoji1), str(emoji2)]
+
+        try:
+            game = Game(self.bot, ctx.channel, ctx.author, user, size=board_size)
+            self.games.append(game)
+            await game.start_game()
+            self.games.remove(game)
+        except Exception:
+            # End the game in the event of an unforeseen error so the players aren't stuck in a game
+            await ctx.send(f"{ctx.author.mention} {user.mention if user else ''} An error occurred. Game failed")
+            if game in self.games:
+                self.games.remove(game)
+            raise
+
+    @commands.group(
+        invoke_without_command=True,
+        aliases=["4inarow", "connect4", "connectfour", "c4"]
+    )
+    async def connect_four(
+            self,
+            ctx: commands.Context,
+            board_size: int = 7,
+            emoji1: EMOJI_CHECK = ":blue_circle:",
+            emoji2: EMOJI_CHECK = ":red_circle:"
```

Comment:
I'm testing the default values locally, but they raise conversion errors. Do they work for you?

Reply:
Fixed.